packages/data-point/lib/entity-types/entity-request/resolve.js

```diff
@@ -91,25 +92,58 @@ function resolveUrl (acc) {
 module.exports.resolveUrl = resolveUrl
 
 /**
+ * @param {Function} callback
+ * @returns {Function} a new callback, that calls the original one
+ */
+function getRequestPromiseWithDebugging (callback) {
+  const debugModule = require('request-debug')
+  clearModule('request-promise')
+  const rpDebug = require('request-promise')
+  clearModule('request-promise')
+  if (callback) {
+    debugModule(rpDebug, callback)
+  } else {
+    debugModule(rpDebug)
+  }
+  return rpDebug
+}
+
+module.exports.getRequestPromiseWithDebugging = getRequestPromiseWithDebugging
+
+/**
+ * @param {Function} requestFn - default request-promise function
  * @param {Accumulator} acc
+ * @returns {Function} request-promise function, with
+ *                     optional debugging attached
  */
-function inspect (acc) {
+function inspect (requestFn, acc) {
   const paramInspect = acc.params && acc.params.inspect
-
-  if (typeof paramInspect === 'function') {
+  const isFunction = typeof paramInspect === 'function'
```

Comment:
can we prevent paramsInspect from being called twice? 

it should be called on its simple form (only `acc` passed) only when params.requestDebug is not true

Reply:
done

Reply:
If we're doing that, we might as well have 2 separate functions (`inspect` and `requestDebugCallback`) - then we get the power of both, but we can also use them simultaneously if necessary.

If we stick with the current approach (using the `params.requestDebug` flag) I like the idea that it *extends* the behavior of `inspect` rather than changing it completely. So, no matter what value `requestDebug` has, `inspect` will be called at least once, synchronously, with the `acc` as the parameter. I think that makes the API more stable

Reply:
It seems like having a separate `requestDebug` function might be the best solution - it's nice and simple, as compared to this `requestDebug` flag that modifies the behavior of a separate setting.